src/disputables/index.js

```diff
@@ -0,0 +1,261 @@
+
+const cachedDescriptions = new Map([])
+
+// Mapping of all disputable apps appId to their
+// corresponding method for describing a disputed action.
+// TODO: Add Conviction Voting
+const DISPUTABLE_ACTIONS = new Map([
+  [
+    DANDELION_VOTING_APP_ID,
+    {
+      abi: disputableDandelionVotingAbi,
+      entity: 'vote',
+      fn: 'getVote',
+      scriptPosition: 10,
+    },
+  ],
+  [
+    DELAY_APP_ID,
+    {
+      abi: disputableDelayAbi,
+      entity: 'delay',
+      fn: 'delayedScripts',
+      scriptPosition: 2,
+
+      // disputable position:
+      // scriptPosition: 4,
+    },
+  ],
+  [
+    VOTING_APP_ID,
+    {
+      abi: disputableVotingAbi,
+      entity: 'vote',
+      fn: 'getVote',
+      scriptPosition: 9,
+    },
+  ],
+])
+
+/**
+ * Get disputable long and short description as well as the URL where the disputed action is taking place
+ * @param {String} disputeId Id of the dispute
+ * @param {String} organization Address of the organization where the action is being disputed
+ * @param {String} disputableAddress Address of the disputable app where the disputed action is taking place
+ * @param {String} disputableActionId Disputed action's id relative to the disputable app (e.g. in the context of a disputable voting app, vote #6)
+ * @returns {Object} Object containing the disputed action short and long description as well as the URL.
+ */
+export async function describeDisputedAction(
+  disputeId,
+  organization,
+  disputableAddress,
+  disputableActionId
+) {
+  try {
+    if (cachedDescriptions.has(disputeId)) {
+      return cachedDescriptions.get(disputeId)
+    }
+    // Get disputable appId
+    const disputableContract = getContract(disputableAddress, disputableAbi)
+    const appId = await disputableContract.appId()
+
+    if (DISPUTABLE_ACTIONS.has(appId)) {
+      const { abi, entity, fn, scriptPosition } = DISPUTABLE_ACTIONS.get(appId)
+
+      // Get disputed action script
+      const disputableAppContract = getContract(disputableAddress, abi)
+      const result = await disputableAppContract[fn](disputableActionId)
+      const evmScript = result[scriptPosition]
+
+      // Get long and short description corrsponding to the disputed action.
+      const [
+        disputedActionRadspec,
+        disputedActionText,
+      ] = await describeActionScript(evmScript, organization)
+
+      const disputedActionDescription = {
+        disputedActionText,
+        disputedActionRadspec,
+
+        // Build URL where the disputed action is taking place
+        disputedActionURL: buildDisputedActionUrl(
+          organization,
+          disputableAddress,
+          entity,
+          disputableActionId
+        ),
+      }
+
+      // Cache disputed action description
+      cachedDescriptions.set(disputeId, disputedActionDescription)
+
+      return disputedActionDescription
+    }
+  } catch (err) {
+    console.error('Error describing disputable action', err)
```

Comment:
Should we also return with a description that mentions it errorred?

It might also be useful to log this error to sentry with something like [this](https://github.com/aragon/aragon/blob/master/src/sentry.js#L11) so we can see if this happens a lot to people.

Reply:
> Should we also return with a description that mentions it errorred?

Good question. Currently we are just returning `No description`  as description but maybe makes sense to tell that it actually errored yeah.

Reply:
Probably not worth showing the full error message.

I think something along the lines of "Failed to fetch description" is fine as long as we're receiving alerts (so we can later debug it rather than ask the user specifically for what they saw).